### PR TITLE
envoyfilter: add bootstrap patch

### PR DESCRIPTION
**Pull request by kyessenov**
Issue https://github.com/istio/istio/issues/26830
Ref https://github.com/istio/istio/pull/32496
Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: networking/v1alpha3/envoy_filter.proto [hunk 1]
```diff
@@ -370,6 +370,38 @@ import "networking/v1alpha3/sidecar.proto";
 //                   cluster_name: xds-grpc
 //           type_urls: ["envoy.extensions.filters.http.wasm.v3.Wasm"]
 // ```
+//
+// The following example adds a Wasm service extension for all proxies using a locally available Wasm file.
```
Comment (louiscryan) [inline networking/v1alpha3/envoy_filter.proto:374]: Could you contextualize the example a bit better? Provide some situations where this would be useful

**Comment by louiscryan** [comment]
Will this be covered in the dedicated WASM API proposal?

File: networking/v1alpha3/envoy_filter.proto (hunk 1 above)
Comment (kyessenov) [reply networking/v1alpha3/envoy_filter.proto:374]: Added two examples. Rate limit and a user database.

**Comment by kyessenov** [comment]
@louiscryan No, Wasm singletons are not in Wasm API, since we made a decision to leave it out of scope to get some MVP first. EnvoyFilter is an escape hatch.

**Comment by louiscryan** [comment]
> @louiscryan No, Wasm singletons are not in Wasm API, since we made a decision to leave it out of scope to get some MVP first. EnvoyFilter is an escape hatch.

Presumably though we would add support eventually and the escape hatch would disappear

**Review by louiscryan — APPROVED** [review]
LGTM

**Comment by kyessenov** [comment]
@louiscryan Correct, we do not have enough evidence to warrant a concrete API for this, but it's better experience with EnvoyFilter than without it for it.

**Comment by mandarjog** [comment]
And it is not just the configuration API.  In one type of usage we may be able to replace configuration driven singleton pattern with a runtime singleton.

**Comment by mandarjog** [comment]
@smawson @nrjpoddar ?
We need two approvals from toc .

**Review by smawson — APPROVED** [review]

**Pull request merged**